src/mpi/init/mpir_init.c

```diff
@@ -192,6 +192,13 @@ int MPII_Init_thread(int *argc, char ***argv, int user_required, int *provided,
     MPIR_ThreadInfo.isThreaded = 0;
 #endif
 
+    /* Initialize gpu in mpl in order to support shm gpu module initialization
+     * inside MPID_Init */
+    if (MPIR_CVAR_ENABLE_GPU) {
+        mpi_errno = MPL_gpu_init();
+        MPIR_ERR_CHECK(mpi_errno);
```

Comment:
Previous, we used
```
MPIR_ERR_CHKANDJUMP(mpl_err != MPL_SUCCESS, mpi_errno, MPI_ERR_OTHER, "**gpu_init");
```
to check this return code. I think using `MPIR_ERR_CHECK` is not the right thing to use here. Same for the `MPL_gpu_finalize();` call.